pkg/ext/stores/kubeconfig/store.go

```diff
@@ -0,0 +1,1487 @@
+	}
+
+	// Note: Status.Value contains tokens' secret keys and mustn't be persisted.
+	kubeconfig.Status.Value = v1Config
+
+	return kubeconfig, nil
+}
+
+// toConfigMap converts a Kubeconfig object to a ConfigMap object.
+func (s *Store) toConfigMap(kubeconfig *ext.Kubeconfig) (*corev1.ConfigMap, error) {
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: *kubeconfig.ObjectMeta.DeepCopy(),
+		Data:       make(map[string]string),
+	}
+	configMap.Namespace = namespace
+	configMap.UID = ""
+
+	if configMap.Annotations == nil {
+		configMap.Annotations = make(map[string]string)
+	}
+	configMap.Annotations[UIDAnnotation] = string(kubeconfig.UID)
+
+	if configMap.Labels == nil {
+		configMap.Labels = make(map[string]string)
+	}
+	configMap.Labels[KindLabel] = KindLabelValue
+
+	if len(kubeconfig.Spec.Clusters) > 0 {
+		serialized, err := json.Marshal(kubeconfig.Spec.Clusters)
+		if err != nil {
+			return nil, fmt.Errorf("error serializing spec.clusters: %w", err)
+		}
+		configMap.Data[ClustersField] = string(serialized)
+	}
+
+	configMap.Data[CurrentContextField] = kubeconfig.Spec.CurrentContext
+	configMap.Data[DescriptionField] = kubeconfig.Spec.Description
+	configMap.Data[TTLField] = strconv.FormatInt(kubeconfig.Spec.TTL, 10)
+
+	// Note: Value should never be persisted!
+	configMap.Data[StatusSummaryField] = kubeconfig.Status.Summary
+	if len(kubeconfig.Status.Conditions) > 0 {
+		serialized, err := json.Marshal(kubeconfig.Status.Conditions)
+		if err != nil {
+			return nil, fmt.Errorf("error serializing status.conditions: %w", err)
+		}
+		configMap.Data[StatusConditionsField] = string(serialized)
+	}
+	if len(kubeconfig.Status.Tokens) > 0 {
+		serialized, err := json.Marshal(kubeconfig.Status.Tokens)
+		if err != nil {
+			return nil, fmt.Errorf("error serializing status.tokens: %w", err)
+		}
+		configMap.Data[StatusTokensField] = string(serialized)
+	}
+
+	return configMap, nil
+}
+
+// fromConfigMap converts a ConfigMap object to a Kubeconfig object.
+func (s *Store) fromConfigMap(configMap *corev1.ConfigMap) (*ext.Kubeconfig, error) {
+	kubeconfig := &ext.Kubeconfig{
+		ObjectMeta: *configMap.ObjectMeta.DeepCopy(),
+		Spec: ext.KubeconfigSpec{
+			Description:    configMap.Data[DescriptionField],
+			CurrentContext: configMap.Data[CurrentContextField],
+		},
+	}
+	kubeconfig.Namespace = ""            // Kubeconfig is not namespaced.
+	delete(kubeconfig.Labels, KindLabel) // Remove an internal label.
+
+	if kubeconfig.Annotations != nil {
+		uid, ok := kubeconfig.Annotations[UIDAnnotation]
+
+		if ok {
+			kubeconfig.UID = types.UID(uid)
+			delete(kubeconfig.Annotations, UIDAnnotation) // Remove an internal annotation.
+		}
+	}
+
+	ttl, err := strconv.ParseInt(configMap.Data[TTLField], 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing TTL for %s: %w", configMap.Name, err)
+	}
+	kubeconfig.Spec.TTL = ttl
+
+	if seriliazed := configMap.Data[ClustersField]; seriliazed != "" {
```

Comment:
```suggestion
	if serialized := configMap.Data[ClustersField]; serialized != "" {
```

Reply:
Thanks for bringing my attention to this as I've noticed a bigger issue on the next line L779 😄